app/common/services/data-export.service.js

```diff
@@ -1,92 +1,68 @@
+    function startPolling(job) {
+        return $interval(function () {
+            // TODO: Question: Should we also delete the job after the job is done/failed?
+            ExportJobEndpoint.getFresh({id: job.id}).$promise.then(function (response) {
+                if (response.status === 'done') {
+                    // when job is done, we stop the polling...
+                    $interval.cancel(polling);
+                    $rootScope.$broadcast('event:export_job:stopped');
+                    // ..and download the file
+                    downloadFile(response.url);
+                    // finally we notify the user
+                    loadingStatus(false);
+                }
+            }, function (err) {
+                    // if there is an error while exporting we stop the polling
+                    $interval.cancel(polling);
+                    $rootScope.$broadcast('event:export_job:stopped');
+                    // and notify the user
+                    Notify.apiErrors(err);
+                }
+            );
+        }, CONST.EXPORT_POLLING_INTERVAL, CONST.EXPORT_POLLING_COUNT);
     }
 
-    function handleArrayBuffer(filename, data, type) {
-        /**
-         * If we have the HTML5 Api for File available we use that. If not, a Blob
-         */
-        function createCSVFile() {
-            if (_.isFunction(File)) {
-                return new File([data], filename, { type: type });
-            } else {
-                return new Blob([data], { type: type });
-            }
-        }
-
-        var blob = createCSVFile();
-
-        if (!_.isUndefined($window.navigator.msSaveBlob)) {
-            /** IE specific workaround for "HTML7007"
-             * https://stackoverflow.com/questions/20310688/blob-download-not-working-in-ie
-            **/
-            $window.navigator.msSaveBlob(blob, filename);
-        } else {
-            var URL = $window.URL || $window.webkitURL;
-            var downloadUrl = URL.createObjectURL(blob);
-            if (filename) {
-                // use HTML5 a[download] attribute to specify filename
-                // Create anchor link
-                var anchor = angular.element('<a/>');
-                anchor.attr({
-                    href: downloadUrl,
-                    download: filename
-                });
-                angular.element(document.body).append(anchor);
-                anchor[0].click();
-                anchor[0].remove();
-            } else {
-                $location.url(downloadUrl);
-            }
-
-            $timeout(function () {
-                URL.revokeObjectURL(downloadUrl);
-            }, 100); // cleanup
+    function cancelExport(jobId) {
+        ExportJobEndpoint.delete({id: jobId});
+        $interval.cancel(polling);
+        $rootScope.$broadcast('event:export_job:stopped');
+        Notify.notify('<p translate="notify.export.canceled_job"></p>');
+    }
 
-        }
+    function downloadFile(downloadUrl) {
```

Comment:
I trigger the download programmatically, do we rather want to give the user a link?